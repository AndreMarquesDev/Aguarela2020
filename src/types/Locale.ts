export enum Locale {
    Pt = 'pt',
    En = 'en',
}
