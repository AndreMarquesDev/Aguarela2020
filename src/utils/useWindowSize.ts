import { useEffect, useState } from 'react';

type WindowSize = {
    width: number;
    height: number;
};

export const useWindowSize = (): WindowSize => {
    const [windowSize, setWindowSize] = useState<WindowSize>({
        width: 0,
        height: 0,
    });

    useEffect(() => {
        const handleResize = (): void => {
            // eslint-disable-next-line react-hooks-extra/no-direct-set-state-in-use-effect
            setWindowSize({
                width: window.innerWidth,
                height: window.innerHeight,
            });
        };

        window.addEventListener('resize', handleResize);

        handleResize();

        return (): void => window.removeEventListener('resize', handleResize);
    }, []);

    return windowSize;
};

export enum Breakpoint {
    Desktop = 1280,
    Phablet = 1023,
    Tablet = 767,
    Mobile = 511,
}
