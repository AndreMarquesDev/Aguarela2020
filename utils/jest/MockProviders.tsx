/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../locales';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = 'pt',
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === 'pt' ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
