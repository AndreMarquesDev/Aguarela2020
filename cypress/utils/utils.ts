import { Locale } from '../../types/Locale';
import {
    menuIconButtonDataTestId,
    nukaCarouselNextButtonDataTestId,
} from '../../utils/dataTestIds';
import { TextsInterface, textsPt, textsEn } from '../../utils/texts';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === Locale.Pt) {
        return textsPt;
    }

    return textsEn;
};

export const openMenuMobile = (isMobile: boolean, isFirefox: boolean): void => {
    if (isMobile) {
        cy.getByDataTestId(menuIconButtonDataTestId).click({ force: isFirefox });
    }
};

export const clickNextArrowButtonIfMobile = (
    isMobile: boolean,
    container: string,
    containerTitle: string
): void => {
    const isFirefox = Cypress.isBrowser('firefox');

    if (isMobile) {
        cy.getByDataTestIdParent(container, nukaCarouselNextButtonDataTestId).click({
            force: isFirefox,
        });

        cy.getByText(container, containerTitle).scrollIntoView();
    }
};
