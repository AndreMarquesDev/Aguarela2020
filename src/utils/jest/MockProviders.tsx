/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
