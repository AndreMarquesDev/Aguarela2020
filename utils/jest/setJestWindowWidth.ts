import { Breakpoint } from '../useWindowSize';

export const setJestWindowWidth = (breakpoint: Breakpoint): void => {
    let width;

    if (breakpoint === Breakpoint.Mobile) {
        width = Breakpoint.Mobile - 1;
    } else if (breakpoint === Breakpoint.Phablet) {
        width = Breakpoint.Phablet - 1;
    } else {
        width = Breakpoint.Desktop + 1;
    }

    Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: width,
    });
};
