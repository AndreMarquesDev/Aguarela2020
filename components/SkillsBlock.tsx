import React, { FC, useContext } from 'react';
import TextsContext from './context/TextsContext';
import Title from './Title';
import { mobileBreakpoint, tabletBreakpoint, useWindowSize } from '../utils/useWindowSize';
import NukaCarousel from './NukaCarousel';
import SkillItem from './SkillItem';

const SkillsBlock: FC = () => {
    const { texts } = useContext(TextsContext);
    const windowSize = useWindowSize();
    const isDesktop = windowSize.width > tabletBreakpoint;
    const isMobile = windowSize.width < mobileBreakpoint;

    return (
        <>
            <section className="container">
                <div className="wrapper genericMargins">
                    <Title colored text={texts.skills} />
                    {isDesktop ? (
                        <ul>
                            <li>
                                <SkillItem description={texts.makingAuditsAndAnalysis} icon="strategy" title={texts.socialMediaStrategy} />
                            </li>
                            <li>
                                <SkillItem description={texts.weCanHelpYourTeam} icon="phone" title={texts.socialMediaConsulting} />
                            </li>
                            <li>
                                <SkillItem description={texts.whenWeSendAMessage} icon="chat" title={texts.communityManagement} />
                            </li>
                            <li>
                                <SkillItem description={texts.planningAndImplementing} icon="money" title={texts.paidSocialAndSearch} />
                            </li>
                            <li>
                                <SkillItem description={texts.measuringResults} icon="graph" title={texts.optimizationAndAnalysis} />
                            </li>
                            <li>
                                <SkillItem description={texts.createAttractiveContent} icon="camera" title={texts.contentCreation} />
                            </li>
                        </ul>
                    ) : (
                        <NukaCarousel width={isMobile ? '100%' : '90%'}>
                            <div className="carouselItem">
                                <SkillItem description={texts.makingAuditsAndAnalysis} icon="strategy" title={texts.socialMediaStrategy} />
                            </div>
                            <div className="carouselItem">
                                <SkillItem description={texts.weCanHelpYourTeam} icon="phone" title={texts.socialMediaConsulting} />
                            </div>
                            <div className="carouselItem">
                                <SkillItem description={texts.whenWeSendAMessage} icon="chat" title={texts.communityManagement} />
                            </div>
                            <div className="carouselItem">
                                <SkillItem description={texts.planningAndImplementing} icon="money" title={texts.paidSocialAndSearch} />
                            </div>
                            <div className="carouselItem">
                                <SkillItem description={texts.measuringResults} icon="graph" title={texts.optimizationAndAnalysis} />
                            </div>
                            <div className="carouselItem">
                                <SkillItem description={texts.createAttractiveContent} icon="camera" title={texts.contentCreation} />
                            </div>
                        </NukaCarousel>
                    )}
                </div>
            </section>

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    .container {
                        background: $white;

                        .wrapper {
                            display: flex;
                            flex-direction: column;
                            align-items: center;
                        }

                        ul {
                            display: grid;
                            grid-template-columns: repeat(3, 1fr);
                            gap: 50rem 4%;

                            @include tablet {
                                grid-template-columns: repeat(2, 1fr);
                            }

                            li {
                                display: flex;
                                flex-direction: column;
                            }
                        }

                        .carouselItem {
                            display: flex;
                            flex-direction: column;
                            align-items: center;
                            text-align: center;
                        }
                    }

                `}
            </style>
        </>
    );
};

export default SkillsBlock;
