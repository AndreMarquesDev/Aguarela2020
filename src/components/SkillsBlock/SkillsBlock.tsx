import React, { useContext } from 'react';
import { tabletMediaQuery } from '../../styles/mediaQueries';
import {
    skillsBlockItemCarouselDataTestId,
    skillsBlockItemWrapperDataTestId,
    skillsBlockSectionDataTestId,
} from '../../utils/dataTestIds';
import { Breakpoint, useWindowSize } from '../../utils/useWindowSize';
import { TextsContext } from '../context/TextsContext';
import { NukaCarousel } from '../NukaCarousel/NukaCarousel';
import { SkillItem } from '../SkillItem/SkillItem';
import { Title } from '../Title/Title';

export const SkillsBlock = (): JSX.Element => {
    const { texts } = useContext(TextsContext);
    const windowSize = useWindowSize();
    const isDesktop = windowSize.width > Breakpoint.Tablet;
    const isMobile = windowSize.width < Breakpoint.Mobile;

    return (
        <>
            <section className="container" data-testid={skillsBlockSectionDataTestId}>
                <div className="wrapper genericMargins">
                    <Title colored text={texts.skills} />
                    {isDesktop
                        ? (
                                <ul>
                                    <li data-testid={skillsBlockItemWrapperDataTestId}>
                                        <SkillItem
                                            description={texts.makingAuditsAndAnalysis}
                                            icon="strategy"
                                            title={texts.socialMediaStrategy}
                                        />
                                    </li>
                                    <li data-testid={skillsBlockItemWrapperDataTestId}>
                                        <SkillItem
                                            description={texts.weCanHelpYourTeam}
                                            icon="phone"
                                            title={texts.socialMediaConsulting}
                                        />
                                    </li>
                                    <li data-testid={skillsBlockItemWrapperDataTestId}>
                                        <SkillItem
                                            description={texts.whenWeSendAMessage}
                                            icon="chat"
                                            title={texts.communityManagement}
                                        />
                                    </li>
                                    <li data-testid={skillsBlockItemWrapperDataTestId}>
                                        <SkillItem
                                            description={texts.planningAndImplementing}
                                            icon="money"
                                            title={texts.paidSocialAndSearch}
                                        />
                                    </li>
                                    <li data-testid={skillsBlockItemWrapperDataTestId}>
                                        <SkillItem
                                            description={texts.measuringResults}
                                            icon="graph"
                                            title={texts.optimizationAndAnalysis}
                                        />
                                    </li>
                                    <li data-testid={skillsBlockItemWrapperDataTestId}>
                                        <SkillItem
                                            description={texts.createAttractiveContent}
                                            icon="camera"
                                            title={texts.contentCreation}
                                        />
                                    </li>
                                </ul>
                            )
                        : (
                                <NukaCarousel width={isMobile ? '100%' : '90%'}>
                                    <div
                                        className="carouselItem"
                                        data-testid={skillsBlockItemCarouselDataTestId}
                                    >
                                        <SkillItem
                                            description={texts.makingAuditsAndAnalysis}
                                            icon="strategy"
                                            title={texts.socialMediaStrategy}
                                        />
                                    </div>
                                    <div
                                        className="carouselItem"
                                        data-testid={skillsBlockItemCarouselDataTestId}
                                    >
                                        <SkillItem
                                            description={texts.weCanHelpYourTeam}
                                            icon="phone"
                                            title={texts.socialMediaConsulting}
                                        />
                                    </div>
                                    <div
                                        className="carouselItem"
                                        data-testid={skillsBlockItemCarouselDataTestId}
                                    >
                                        <SkillItem
                                            description={texts.whenWeSendAMessage}
                                            icon="chat"
                                            title={texts.communityManagement}
                                        />
                                    </div>
                                    <div
                                        className="carouselItem"
                                        data-testid={skillsBlockItemCarouselDataTestId}
                                    >
                                        <SkillItem
                                            description={texts.planningAndImplementing}
                                            icon="money"
                                            title={texts.paidSocialAndSearch}
                                        />
                                    </div>
                                    <div
                                        className="carouselItem"
                                        data-testid={skillsBlockItemCarouselDataTestId}
                                    >
                                        <SkillItem
                                            description={texts.measuringResults}
                                            icon="graph"
                                            title={texts.optimizationAndAnalysis}
                                        />
                                    </div>
                                    <div
                                        className="carouselItem"
                                        data-testid={skillsBlockItemCarouselDataTestId}
                                    >
                                        <SkillItem
                                            description={texts.createAttractiveContent}
                                            icon="camera"
                                            title={texts.contentCreation}
                                        />
                                    </div>
                                </NukaCarousel>
                            )}
                </div>
            </section>

            <style jsx>
                {`
                    .container {
                        background: var(--white);

                        .wrapper {
                            display: flex;
                            flex-direction: column;
                            align-items: center;
                        }

                        ul {
                            display: grid;
                            grid-template-columns: repeat(3, 1fr);
                            gap: 50rem 4%;

                            @media (${tabletMediaQuery}) {
                                grid-template-columns: repeat(2, 1fr);
                            }

                            li {
                                display: flex;
                                flex-direction: column;
                            }
                        }

                        .carouselItem {
                            display: flex;
                            flex-direction: column;
                            align-items: center;
                            text-align: center;
                        }
                    }
                `}
            </style>
        </>
    );
};
