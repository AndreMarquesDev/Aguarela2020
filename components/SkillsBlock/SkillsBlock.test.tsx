import '@testing-library/jest-dom';
import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import { textsPt, textsEn } from '../../utils/texts';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';
import { SkillsBlock } from './SkillsBlock';
import { Breakpoint } from '../../utils/useWindowSize';
import {
    skillsBlockItemWrapperDataTestId,
    skillsBlockItemCarouselDataTestId,
} from '../../utils/dataTestIds';
import { MockProviders } from '../../utils/jest/MockProviders';
import { Locale } from '../../types/Locale';

const renderComponent = (language: Locale = Locale.Pt): RenderResult => {
    return render(
        <MockProviders language={language}>
            <SkillsBlock />
        </MockProviders>
    );
};

describe('<SkillsBlock />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        const title = screen.getByText(textsPt.skills);
        const block1Title = screen.getByText(textsPt.socialMediaStrategy);
        const block2Title = screen.getByText(textsPt.socialMediaConsulting);
        const block3Title = screen.getByText(textsPt.communityManagement);
        const block4Title = screen.getByText(textsPt.paidSocialAndSearch);
        const block5Title = screen.getByText(textsPt.optimizationAndAnalysis);
        const block6Title = screen.getByText(textsPt.contentCreation);

        const numberOfBlocks = screen.getAllByTestId(skillsBlockItemWrapperDataTestId);
        const carouselItemWrapper = screen.queryByTestId(skillsBlockItemCarouselDataTestId);

        expect(title).toBeInTheDocument();
        expect(block1Title).toBeInTheDocument();
        expect(block2Title).toBeInTheDocument();
        expect(block3Title).toBeInTheDocument();
        expect(block4Title).toBeInTheDocument();
        expect(block5Title).toBeInTheDocument();
        expect(block6Title).toBeInTheDocument();

        expect(numberOfBlocks.length).toBe(6);
        expect(carouselItemWrapper).not.toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = renderComponent(Locale.En);

        const title = screen.getByText(textsEn.skills);
        const block1Title = screen.getByText(textsEn.socialMediaStrategy);
        const block2Title = screen.getByText(textsEn.socialMediaConsulting);
        const block3Title = screen.getByText(textsEn.communityManagement);
        const block4Title = screen.getByText(textsEn.paidSocialAndSearch);
        const block5Title = screen.getByText(textsEn.optimizationAndAnalysis);
        const block6Title = screen.getByText(textsEn.contentCreation);

        expect(title).toBeInTheDocument();
        expect(block1Title).toBeInTheDocument();
        expect(block2Title).toBeInTheDocument();
        expect(block3Title).toBeInTheDocument();
        expect(block4Title).toBeInTheDocument();
        expect(block5Title).toBeInTheDocument();
        expect(block6Title).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly on tablet', () => {
        setJestWindowWidth(Breakpoint.Tablet);

        const { container } = renderComponent();

        const carouselItemWrapper = screen.getAllByTestId(skillsBlockItemCarouselDataTestId);
        const numberOfBlocks = screen.queryByTestId(skillsBlockItemWrapperDataTestId);

        expect(carouselItemWrapper.length).toBe(6);
        expect(numberOfBlocks).not.toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly on mobile', () => {
        setJestWindowWidth(Breakpoint.Mobile);

        const { container } = renderComponent();

        const carouselItemWrapper = screen.getAllByTestId(skillsBlockItemCarouselDataTestId);
        const numberOfBlocks = screen.queryByTestId(skillsBlockItemWrapperDataTestId);

        expect(carouselItemWrapper.length).toBe(6);
        expect(numberOfBlocks).not.toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
