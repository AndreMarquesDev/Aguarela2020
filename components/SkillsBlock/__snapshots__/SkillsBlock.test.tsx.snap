// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkillsBlock /> renders properly 1`] = `
<div>
  <section
    class="jsx-2589310862 container"
  >
    <div
      class="jsx-2589310862 wrapper genericMargins"
    >
      <h1
        class="jsx-3155756322 colored"
      >
        Skills
      </h1>
      <ul
        class="jsx-2589310862"
      >
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Estratégia de Redes Sociais"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Estratégia de Redes Sociais
          </strong>
          <p
            class="jsx-1484428706"
          >
            Realizar uma auditoria e análise detalhada, com base nos insights e objetivos da marca, definindo em que redes sociais a marca deve estar presente e quais as oportunidades disponíveis.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Consultoria de Redes Sociais"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Consultoria de Redes Sociais
          </strong>
          <p
            class="jsx-1484428706"
          >
            Posso ajudar a sua equipa na forma como a marca está presente nas redes sociais, como comunica, como alcançar os objetivos pretendidos e como aproveitar as oportunidades disponíveis no mercado digital do seu negócio.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Gestão de Comunidade"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Gestão de Comunidade
          </strong>
          <p
            class="jsx-1484428706"
          >
            Quando enviamos uma mensagem aos nossos amigos e eles demoram dias a responder-nos não é muito agradável e o mesmo acontece com as marcas. Faço gestão de comunidade, respondendo a mensagens e comentários da sua audiência em prazos toleráveis e ideais.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Paid Social e Search"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Paid Social e Search
          </strong>
          <p
            class="jsx-1484428706"
          >
            Planeio, implemento e acompanho campanhas paid social (redes sociais) e campanhas paid search (anúncios no Google), otimizando os objetivos de marketing da sua marca ou negócio.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Análise e Otimização"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Análise e Otimização
          </strong>
          <p
            class="jsx-1484428706"
          >
            Medir resultados de forma a otimizar a estratégia é a chave para uma presença de sucesso nas redes sociais. Em todos os meus projetos faço uma análise mensal detalhada de forma a existir uma adaptação contínua ao mercado e às exigências dos consumidores.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Criação de Conteúdo"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Criação de Conteúdo
          </strong>
          <p
            class="jsx-1484428706"
          >
            Crio conteúdo (fotográfico, animado, design e vídeo) atrativo e relevante para os consumidores. Ao reunir insights de consumidores e know-how consigo saber o tipo de conteúdo que melhor funciona para cada negócio.
          </p>
        </li>
      </ul>
    </div>
  </section>
</div>
`;

exports[`<SkillsBlock /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-2589310862 container"
  >
    <div
      class="jsx-2589310862 wrapper genericMargins"
    >
      <h1
        class="jsx-3155756322 colored"
      >
        Skills
      </h1>
      <ul
        class="jsx-2589310862"
      >
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Social Media strategy"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Social Media strategy
          </strong>
          <p
            class="jsx-1484428706"
          >
            Making audits and detailed analysis, based on the brand's insights and goals, deciding in what social networks the brand should be present in and what are the available opportunities.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Social Media consulting"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Social Media consulting
          </strong>
          <p
            class="jsx-1484428706"
          >
            I can help your team in the way your brand is present on social media, how it communicates, how to achieve the intended goals and how to take advantage of the opportunities available in the digital market of your business.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Community Management"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Community Management
          </strong>
          <p
            class="jsx-1484428706"
          >
            When we send a message to our friends and they take days to answer us it is not very pleasant and the same is true with brands. I do community management, responding to messages and comments from your audience within tolerable and ideal deadlines.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Paid Social and Search"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Paid Social and Search
          </strong>
          <p
            class="jsx-1484428706"
          >
            I plan, implement and monitor paid social campaigns (social media) and paid search campaigns (ads on Google), optimizing the marketing objectives of your brand or business.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Optimization and Analysis"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Optimization and Analysis
          </strong>
          <p
            class="jsx-1484428706"
          >
            Measuring results in order to optimize the strategy is the key to a successful presence on social media. In all my projects I do a detailed monthly analysis so that there is a continuous adaptation to the market and to the demands of consumers.
          </p>
        </li>
        <li
          class="jsx-2589310862"
          data-testid="skillsBlock_skillItemWrapper"
        >
          <div
            style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 160px; height: 130px;"
          >
            <img
              alt="Content creation"
              class="imageOpacity"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
          <strong
            class="jsx-1484428706"
          >
            Content creation
          </strong>
          <p
            class="jsx-1484428706"
          >
            I create attractive and relevant content to consumers (photographic, animated, design and video). By gathering consumer insights and know-how, I know the type of content that works best for each business.
          </p>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
